pkg/prometheus/resource_selector.go

```diff
@@ -677,6 +677,24 @@ func (rs *ResourceSelector) SelectScrapeConfigs(ctx context.Context, listFn List
 			}
 		}
 
+		var scrapeInterval, scrapeTimeout monitoringv1.Duration
```

Comment:
WDYT about having an initial value as empty and reducing the cyclomatic complexity by removing the `else` statement?

```suggestion
		var scrapeInterval, scrapeTimeout monitoringv1.Duration = "", ""
		
		if sc.Spec.ScrapeInterval != nil {
		  scrapeInterval = *sc.Spec.ScrapeInterval
		}
		
		if sc.Spec.ScrapeTimeout != nil {
          scrapeTimeout = *sc.Spec.ScrapeTimeout
        }
```